Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,467 @@
+            set
+            {
+                if (surfaceNormalOffset != value)
+                {
+                    surfaceNormalOffset = value;
+                }
+            }
+        }
+
+        [SerializeField]
+        [Tooltip("If true, the GameObjectToPlace will remain upright and in line with Vector3.up")]
+        private bool keepOrientationVertical = false;
+
+        /// <summary>
+        /// If true, the GameObjectToPlace will remain upright and in line with Vector3.up
+        /// </summary>
+        public bool KeepOrientationVertical
+        {
+            get => keepOrientationVertical;
+            set
+            {
+                if (keepOrientationVertical != value)
+                {
+                    keepOrientationVertical = value;
+                }
+            }
+        }
+
+
+        [SerializeField]
+        [Tooltip("If true, the spatial mesh will be visible while the object is in the placing state.")]
+        private bool spatialMeshVisible = true;
+
+        /// <summary>
+        /// If true, the spatial mesh will be visible while the object is in the placing state.
+        /// </summary>
+        public bool SpatialMeshVisible
+        {
+            get => spatialMeshVisible;
+            set
+            {
+                if (spatialMeshVisible != value)
+                {
+                    spatialMeshVisible = value; 
+                }
+            }
+        }
+
+
+
+        [SerializeField]
+        [Tooltip("")]
+        private SpatialAwarenessMeshDisplayOptions initialSpatialMeshDisplayOption;
+
+        /// <summary>
+        /// Get the spatial mesh visibility on start.  If the the spatial mesh is visible by default then do not toggle the
+        /// spatial mesh visibility while placing the game object.  Spatial mesh visibility refers to if the wire frame material is
+        /// on the spatial mesh or not.
+        /// </summary>
+        public SpatialAwarenessMeshDisplayOptions InitialSpatialMeshDisplayOption
+        {
+            get
+            {
+
+                IReadOnlyList<IMixedRealitySpatialAwarenessMeshObserver> observers = spatialAwarenessSystemAcecess.GetDataProviders<IMixedRealitySpatialAwarenessMeshObserver>();
+
+                foreach (IMixedRealitySpatialAwarenessMeshObserver observer in observers)
+                {
+                    return observer.DisplayOption;
+                }
+
+                // if the spatial awareness is not enabled
+                return SpatialAwarenessMeshDisplayOptions.None;   
+            }
+        }
+
+        private IMixedRealityDataProviderAccess spatialAwarenessSystemAcecess;
+
+        [SerializeField]
+        [Tooltip("If false, the game object to place will not change its rotation according to the surface hit.  The object will" +
+            "remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal" +
+            "if there is a hit.")]
```

Comment:
Same comment on needing spaces after/before the ends/beginnings of lines so that words don't get smushed

Reply:
Spaces added